src/promptflow/tests/sdk_cli_test/e2etests/test_custom_strong_type_connection.py

```diff
@@ -0,0 +1,152 @@
+
+    def test_connection_update(self):
+        name = f"Connection_{str(uuid.uuid4())[:4]}"
+        conn = MyCustomConnection(name=name, api_key="test", api_base="test")
+        # Create
+        _client.connections.create_or_update(conn)
+        # Get
+        custom_conn = _client.connections.get(name)
+        assert pydash.omit(custom_conn._to_dict(), ["created_date", "last_modified_date", "name"]) == {
+            "module": "promptflow.connections",
+            "type": "custom",
+            "configs": {
+                "api_base": "test",
+                "azureml.connection.custom_type": "MyCustomConnection",
+                "azureml.connection.module": "sdk_cli_test.e2etests.test_custom_strong_type_connection",
+            },
+            "secrets": {"api_key": "******"},
+        }
+        # Update
+        custom_conn.api_base = "test2"
+        result = _client.connections.create_or_update(custom_conn)
+        assert pydash.omit(result._to_dict(), ["created_date", "last_modified_date", "name"]) == {
+            "module": "promptflow.connections",
+            "type": "custom",
+            "configs": {
+                "api_base": "test2",
+                "azureml.connection.custom_type": "MyCustomConnection",
+                "azureml.connection.module": "sdk_cli_test.e2etests.test_custom_strong_type_connection",
+            },
+            "secrets": {"api_key": "******"},
+        }
+        # List
+        result = _client.connections.list()
+        assert len(result) > 0
+        # Delete
+        _client.connections.delete(name)
+        with pytest.raises(Exception) as e:
+            _client.connections.get(name)
+        assert "is not found." in str(e.value)
+
+    def test_connection_get_and_update(self):
+        # Test api key not updated
+        name = f"Connection_{str(uuid.uuid4())[:4]}"
+        conn = MyCustomConnection(name=name, api_key="test", api_base="test")
+        result = _client.connections.create_or_update(conn)
+        assert result.api_key == SCRUBBED_VALUE
+        # Update api_base only Assert no exception
+        result.api_base = "test2"
+        result = _client.connections.create_or_update(result)
+        assert result._to_dict()["configs"]["api_base"] == "test2"
+        # Assert value not scrubbed
+        assert result._secrets["api_key"] == "test"
+        _client.connections.delete(name)
+        # Invalid update
+        with pytest.raises(Exception) as e:
+            result._secrets = {}
+            _client.connections.create_or_update(result)
+        assert "secrets ['api_key'] value invalid, please fill them" in str(e.value)
+
+    @pytest.mark.parametrize(
+        "file_name, expected_updated_item, expected_secret_item",
+        [
+            ("custom_strong_type_connection.yaml", ("api_base", "new_value"), ("api_key", "<to-be-replaced>")),
+        ],
+    )
+    def test_upsert_connection_from_file(self, file_name, expected_updated_item, expected_secret_item):
```

Comment:
Test if custom connection type not match?

Reply:
Removed temporarily for this PR